StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerDataSource.swift

```diff
@@ -39,7 +39,25 @@ final class LinkAccountPickerDataSourceImplementation: LinkAccountPickerDataSour
     private let apiClient: FinancialConnectionsAPIClient
     private let clientSecret: String
     private let consumerSession: ConsumerSessionData
-    let dataAccessNotice: FinancialConnectionsDataAccessNotice?
+    var dataAccessNotice: FinancialConnectionsDataAccessNotice? {
+        var dataAccessNotice: FinancialConnectionsDataAccessNotice?
+        if
+            let networkedAccountsResponse,
+            let returningNetworkingUserAccountPicker = networkedAccountsResponse.display?.text?.returningNetworkingUserAccountPicker,
+            !returningNetworkingUserAccountPicker.accounts.isEmpty
+        {
+            if selectedAccounts.isEmpty {
+                dataAccessNotice = returningNetworkingUserAccountPicker.accounts.first?.dataAccessNotice
+            } else if selectedAccounts.count == 1 {
+                dataAccessNotice = selectedAccounts.first?.accountPickerAccount.dataAccessNotice
+            } else { // multiple accounts selected
+                dataAccessNotice = returningNetworkingUserAccountPicker.multipleAccountTypesSelectedDataAccessNotice
+            }
+        }
+        return dataAccessNotice ?? consentDataAccessNotice
+    }
```

Comment:
[Stripe.js code for this](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/15a677fc/src/linkedAccounts/inner/components/panes/v3/ConsentPane/content/DataAccessNoticeDrawer/useDataAccessNoticeText.ts#L31-L72)